### PR TITLE
Resolve #942 python3.9 support

### LINKED ISSUE #942: Python 3.9 not supported
```
  File "/usr/local/Cellar/sceptre/2.4.0_1/libexec/lib/python3.9/site-packages/networkx/algorithms/dag.py", line 23, in <module>
    from fractions import gcd
ImportError: cannot import name 'gcd' from 'fractions' (/usr/local/Cellar/python@3.9/3.9.0/Frameworks/Python.framework/Versions/3.9/lib/python3.9/fractions.py)
```

The networkx version specified is too old and does not support Python 3.9. fractions.gcd was removed.

### PR DESCRIPTION
**Pull request by chrcoe**
With python 3.9 networkx library no longer worked on version 2.1

This PR updates to use the newest networkx library.  This addresses https://github.com/Sceptre/sceptre/issues/942

To get the tests passing, upgraded versions of `coverage` and `troposphere` are needed, but I was having trouble getting `troposphere` 2.6.3 installed (the version that supports python 3.9)

That being said, I have use this locally with python 3.9 successfully.  Feel free to decline this PR simply because the test libraries need updated to support 3.9 and this PR does not do that.

## PR Checklist

- [x] Wrote a good commit message & description [see guide below].
- [x] Commit message starts with `[Resolve #issue-number]`.
- [x] Added/Updated unit tests. N/A
- [x] Added/Updated integration tests (if applicable). N/A
- [ ] All unit tests (`make test`) are passing.
- [x] Used the same coding conventions as the rest of the project.
- [x] The new code passes flake8 (`make lint`) checks.
- [x] The PR relates to _only_ one subject with a clear title.
      and description in grammatically correct, complete sentences.

## Approver/Reviewer Checklist

- [ ] Before merge squash related commits.

## Other Information

[Guide to writing a good commit](http://chris.beams.io/posts/git-commit/)

### CODE REVIEW

**Comment by ykhalyavin** [comment]
@ngfgrant  @zaro0508 could you guys please review it, it enables is the compatibility with the latest **cfn-lint**, which gives you an error with sceptre==2.4.0

It's critical for us, since we validate templates with cfn-lint (CI/CD) and we started to use AWS::SSO::* resources which are not supported by older versions of cfn-lint

The error:
`ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.

cfn-lint 0.46.0 requires networkx~=2.4; python_version >= "3.5", but you have networkx 2.1 which is incompatible.`

**Comment by ykhalyavin** [comment]
@chrcoe I would change all networkx>=2.5 to "networkx>=2.5,<3" to make sure if networkx API changes it does not break something

File: requirements/dev.txt [hunk 1]
```diff
@@ -7,7 +7,7 @@ freezegun>=0.3.12,<0.4.0
 Jinja2>=2.8,<3
 mock>=2.0.0,<2.1.0
 moto>=1.3.8,<2.0.0
-networkx==2.1
+networkx>=2.5
```
Comment (zaro0508) [inline requirements/dev.txt:10]: how about we scope to less than 3 as well?  `networkx>=2.5,<3`?

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: requirements/dev.txt (hunk 1 above)
Comment (chrcoe) [reply requirements/dev.txt:10]: good idea

**Review by zaro0508 — APPROVED** [review]

**Comment by lime-green** [comment]
@zaro0508 are you able to merge and cut a new release for this? This has been a pain-point for us

**Comment by zaro0508** [comment]
unfortunately i do not have merge powers @lime-green.  you'll want to ping @ngfgrant for that.

**Review by zaro0508 — CHANGES_REQUESTED** [review]
To make this work I think we first need to setup sceptre to run tests against 3.8 & 3.9 https://github.com/Sceptre/sceptre/blob/master/tox.ini#L2

**Comment by zaro0508** [comment]
closing since this is the same PR as #1008

**Pull request closed without merging**